spec/controller/donut_shop_spec.rb

```diff
@@ -1,25 +1,39 @@
 require 'rails_helper'
 
-RSpec.describe DonutShopsController, :type => :controller do
+RSpec.describe DonutShopsController,  :type => :controller do
+  render_views
 
   let!(:shop) { DonutShop.create!(name: Faker::Lorem.word, description: Faker::Lorem.sentence, address: Faker::Address, phone: Faker::PhoneNumber.phone_number, website: Faker::Internet.url, donut_id: 1 ) }
 
-  describe "GET #show" do
+  describe "GET #index" do
 
     it "renders the index page" do 
-      get :show, id: shop.id
-      expect(response).to render_template(:show)
+      get :index
+      expect(response).to render_template(:index)
     end
 
     it "renders the index page" do 
-      get :show, id: 1
+      get :show, { :id => shop.to_param }
       expect(response.status).to eq(200)
     end
   end
 
-  describe "POST #create" do
+  
+
+  # The test below need work. 
```

Comment:
@BintLopez The post test for Donut Shop aren't passing, while the similar tests for Donuts do.